XMonad/Hooks/StatusBar.hs

```diff
@@ -311,9 +313,19 @@ statusBarPropTo prop cmd pp = def
 statusBarPipe :: String -- ^ The command line to launch the status bar
               -> X PP   -- ^ The pretty printing options
               -> IO StatusBarConfig
-statusBarPipe cmd xpp  = do
-    h <- spawnPipe cmd
-    return $ def { sbLogHook = xpp >>= \pp -> dynamicLogWithPP pp { ppOutput = hPutStrLn h } }
+statusBarPipe cmd xpp = do
+    hRef <- newIORef Nothing
+    return $ def
+        { sbStartupHook = io (writeIORef hRef . Just =<< spawnPipe cmd)
+        , sbLogHook     = do
+            pp <- xpp
+            h' <- io (readIORef hRef)
+            whenJust h' $ \h -> dynamicLogWithPP pp { ppOutput = hPutStrLn h }
```

Comment:
I'd say either indent this by four spaces or indent the rest of the function by two spaces as well